document-translator/script.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './pdf.min.mjs';
+
+const {pdfjsLib} = globalThis;
+pdfjsLib.GlobalWorkerOptions.workerSrc = './pdf.worker.mjs';
+
+const [openButton, exampleButton, languageButton, translateButton] = Array.from(document.querySelectorAll('button'));
+const output = document.querySelector('output');
+const detectedLanguageParagraph = document.querySelector('.detected-language');
+const translationParagraph = document.querySelector('.translation');
+const img = document.querySelector('img');
+
+let useExample = false;
+let detectedLanguage = false;
+
+openButton.addEventListener('click', async () => {
+  output.innerHTML = '';
+  translationParagraph.innerHTML = '';
+  detectedLanguageParagraph.innerHTML = '';
+  detectedLanguage = false;
+  img.src = '';
+
+  let file;
+  if (!useExample) {
+    try {
+      const [handle] = await showOpenFilePicker({
+        types: [
+          {
+            description: 'Image or PDF files',
+            accept: {
+              'image/*': ['.png', '.gif', '.jpeg', '.jpg', '.webp', '.avif'],
+              'application/pdf': ['.pdf'],
+            },
+          },
+        ],
+      });
+      file = await handle.getFile();
+      if (file.type === 'application/pdf') {
+        const pdfBytes = new Uint8Array(await file.arrayBuffer());
+        file = await convertFirstPageToPngBlob(pdfBytes);
+      }
+    } catch (err) {
+      console.error(err.name, err.message);
+    }
+  } else {
+    useExample = false;
+    file = await fetch('specimen.png').then(response => response.blob());
+  }
+  const blobURL = URL.createObjectURL(file);
+  img.src = blobURL;
+  setTimeout(() => {
+    URL.revokeObjectURL(blobURL);
+  }, 0);
+
+  try {
+    const session = await LanguageModel.create({
+      expectedInputs: [{ type: 'image' }, { type: 'text' }],
```

Comment:
FWIW I use `expectedInputs: [{ type: "image" }],` only and it works. I wonder if this is a feature or a bug...

Reply:
It's a feature. `type: 'text'` is the assumed default. I just want to make it explicit and it [might come in handy](https://github.com/webmachinelearning/prompt-api#:~:text=//%20%7B%20type%3A%20%22text%22%20%7D%20is%20not%20necessary%20to%20include%20explicitly%2C%20unless%0A%20%20//%20you%20also%20want%20to%20include%20expected%20input%20languages%20for%20text.).